### PR TITLE
Do not handle changes to fields not in the corresponding DB table

### PR DESCRIPTION
**Pull request by hboomsma**

### CODE REVIEW

File: src/Provider/EntityMutationMetadataProvider.php [hunk 1]
```diff
@@ -66,39 +66,45 @@ public function createOriginalEntity(EntityManagerInterface $em, $entity)
...
+     * @throws \InvalidArgumentException
      */
     public function getMutatedFields(EntityManagerInterface $em, $entity, $original)
     {
         $mutation_data = [];
         /* @var $metadata \Doctrine\ORM\Mapping\ClassMetadata */
         $metadata     = $em->getClassMetadata(get_class($entity));
         $fields       = $metadata->getFieldNames();
-        $associations = $metadata->getAssociationNames();
+        $associations = [];//$metadata->getAssociationNames();
```
Comment (iltar) [inline src/Provider/EntityMutationMetadataProvider.php]: Forgot to remove commented code

File: src/Provider/EntityMutationMetadataProvider.php [hunk 2]
```diff
@@ -107,7 +113,8 @@ public function getMutatedFields(EntityManagerInterface $em, $entity, $original)
                 $association_target_meta,
                 $association_value_left,
                 $association_value_right
-            )) {
+            )
+            ) {
```
Comment (iltar) [inline src/Provider/EntityMutationMetadataProvider.php]: should be `)) {` as far as I know

File: src/Provider/EntityMutationMetadataProvider.php [hunk 3]
```diff
@@ -134,7 +141,7 @@ private function hasAssociationChanged(ClassMetadata $association_meta, $left, $
                 function ($a, $b) {
                     // note that equal returns 0, difference should return -1 or 1
                     if (!is_object($a) && !is_object($b)) {
-                        return (string) $a === (string) $b ? 0 : 1;
+                        return (string)$a === (string)$b ? 0 : 1;
```
Comment (iltar) [inline src/Provider/EntityMutationMetadataProvider.php]: is this a codestyle change?

Comment (hboomsma) [inline src/Provider/EntityMutationMetadataProvider.php]: yes, this is phpcbf

File: src/Provider/EntityMutationMetadataProvider.php [hunk 4]
```diff
@@ -157,37 +164,21 @@ function ($a, $b) {
...
     }
 
     /**
      * @param EntityManagerInterface $em
-     * @param mixed                  $entity
-     * @return bool
+     * @return array
      */
-    public function isEntityManaged(EntityManagerInterface $em, $entity)
+    public function getChanges(EntityManagerInterface $em)
```
Comment (iltar) [inline src/Provider/EntityMutationMetadataProvider.php]: add a small remark that the name has changed (not official public api)

**Comment by iltar** [comment]
Can you mark some of the classes as `@internal`

**Comment by nicoschoenmaker** [comment]
Looks good to me :+1:

**Pull request merged**